torch/_meta_registrations.py

```diff
@@ -1985,7 +1998,10 @@ def activate_meta():
         }:
             pass
         else:
-            _meta_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
+            if op_overload.name().startswith("c10d"):
+                _meta_c10d_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
+            else:
+                _meta_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
```

Comment:
So is the reason for the hack that you're trying to register all of the c10d ops with meta? But if instead you only tried to register traceable_allreduce then it would work?